src/python/src/pyconfigoptions.cpp

```diff
@@ -0,0 +1,289 @@
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "pyconfigoptions.h"
+#include "srexception.h"
+
+using namespace SmartRedis;
+
+namespace py = pybind11;
+
+PyConfigOptions::PyConfigOptions()
+{
+    try {
+        _configoptions = NULL;
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "during ConfigOptions construction.");
+    }
+}
+
+PyConfigOptions::PyConfigOptions(ConfigOptions* configoptions)
+{
+    _configoptions = configoptions;
+}
+
+PyConfigOptions::~PyConfigOptions()
+{
+    if (_configoptions != NULL) {
+        delete _configoptions;
+        _configoptions = NULL;
+    }
+}
+
+ConfigOptions* PyConfigOptions::get() {
+    return _configoptions;
+}
+
+// Instantiate ConfigOptions from environment variables
+PyConfigOptions* PyConfigOptions::create_from_environment(
+    const std::string& db_prefix)
+{
+    try {
+        auto cfgOpts = ConfigOptions::create_from_environment(db_prefix);
+        ConfigOptions* pCfgOpts = cfgOpts.release();
+        return new PyConfigOptions(pCfgOpts);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing create_from_environment.");
+    }
+}
+
+// Configuration via JSON file or JSON blob is anticipated in the future
+// but not supported yet
+#ifdef FUTURE_CONFIG_SUPPORT
```

Comment:
Another quick note to remember to remove

Reply:
done